src/cpu/x64/brgemm/jit_brgemm_kernel.cpp

```diff
@@ -523,11 +523,12 @@ dim_t jit_brgemm_kernel_t<Wmm>::A_offset(
 template <typename Wmm>
 dim_t jit_brgemm_kernel_t<Wmm>::B_offset(
         dim_t ld, dim_t rd, bool is_amx) const noexcept {
-    int typesize_scale = one_of(brg.dt_b, data_type::nf4, data_type::s4, data_type::u4, data_type::f4_e2m1) ? 2 : 1;
+    int typesize_scale = brg.dt_b == data_type::u2 ? 4 : one_of(brg.dt_b, data_type::nf4, data_type::s4, data_type::u4, data_type::f4_e2m1) ? 2 : 1;
```

Comment:
```suggestion
    int typesize_scale = [] {
    if (brg.dt_b == data_type::u2) {
        return 4;
    } else if (one_of(brg.dt_b, data_type::nf4, data_type::s4, data_type::u4, data_type::f4_e2m1)) {
       return 2;
    } else {
       return 1;
    }
} ();
```
Here and in similar ternary expressions, simply to improve readability.

Reply:
Applied. Thanks Maksim!